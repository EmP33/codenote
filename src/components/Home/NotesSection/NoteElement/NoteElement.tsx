import React, { useEffect, useState } from "react";
import { Card, CardContent, Typography, Box } from "@mui/material";
import { motion } from "framer-motion";

interface NoteElementProps {
  title: string;
  date: string;
  content: string;
}

const NoteElement: React.FC<NoteElementProps> = ({ title, date, content }) => {
  const [enableDragging, setEnableDragging] = useState(true);

  useEffect(() => {
    const handleResize = () => {
      if (window.screen.width <= 600) {
        setEnableDragging(false);
      } else {
        setEnableDragging(true);
      }
    };
    window.addEventListener("resize", handleResize);
    handleResize();
  }, []);
  return (
    <Card
      component={motion.div}
      drag={enableDragging}
      dragConstraints={{ left: 0, top: 0, right: 0, bottom: 0 }}
      sx={{
        height: 250,
        width: 350,
        margin: "2rem auto",
        background: "var(--color-secondary-light)",
        color: "#fff",
        border: "2px solid var(--color-tertiary)",
        position: "relative",
        zIndex: 1,
        cursor: "pointer",

        "@media only screen and (max-width: 400px)": {
          width: "auto",
          maxWidth: 280,
        },

        "&:before": {
          content: "''",
          width: "100%",
          height: "70%",
          background: "var(--color-secondary)",
          position: "absolute",
          borderRadius: "0 0 50% 50%",
          top: 0,
          left: 0,
          zIndex: -1,
        },
      }}
    >
      <CardContent>
        <Box
          sx={{ display: "flex", justifyContent: "space-between", zIndex: 5 }}
        >
          <Typography sx={{ fontSize: 18, marginBottom: 3 }} gutterBottom>
            {title}
          </Typography>
          <Typography
            sx={{
              fontSize: 16,
              color: "var(--color-base)",
            }}
          >
            {date}
          </Typography>
        </Box>
        <Typography sx={{ fontSize: 14, width: "75%" }}>{content}</Typography>
      </CardContent>
    </Card>
  );
};

export default NoteElement;
