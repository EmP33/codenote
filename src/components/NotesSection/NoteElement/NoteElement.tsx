import React from "react";
import { Card, CardContent, Typography, Box } from "@mui/material";

interface NoteElementProps {
  title: string;
  date: string;
  content: string;
}

const NoteElement: React.FC<NoteElementProps> = ({ title, date, content }) => {
  return (
    <Card
      sx={{
        height: 250,
        width: 350,
        margin: "2rem auto",
        background: "var(--color-secondary-light)",
        color: "#fff",
        border: "2px solid var(--color-tertiary)",
        position: "relative",
        zIndex: 1,

        "@media only screen and (max-width: 400px)": {
          width: "auto",
          maxWidth: 270,
        },

        "&:before": {
          content: "''",
          width: "100%",
          height: "70%",
          background: "var(--color-secondary)",
          position: "absolute",
          borderRadius: "0 0 50% 50%",
          top: 0,
          left: 0,
          zIndex: -1,
        },
      }}
    >
      <CardContent>
        <Box
          sx={{ display: "flex", justifyContent: "space-between", zIndex: 5 }}
        >
          <Typography sx={{ fontSize: 18, marginBottom: 3 }} gutterBottom>
            {title}
          </Typography>
          <Typography
            sx={{
              fontSize: 16,
              color: "var(--color-base)",
            }}
          >
            {date}
          </Typography>
        </Box>
        <Typography sx={{ fontSize: 14, width: "75%" }}>{content}</Typography>
      </CardContent>
    </Card>
  );
};

export default NoteElement;
