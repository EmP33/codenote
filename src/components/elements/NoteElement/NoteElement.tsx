import React from "react";
import { Card, CardContent, Typography, Box } from "@mui/material";

interface NoteElementProps {
  title: string;
  date: string;
  content: string;
}

const NoteElement: React.FC<NoteElementProps> = ({ title, date, content }) => {
  return (
    <Card
      sx={{
        height: 250,
        width: 350,
        margin: "2rem auto",
        background: "var(--color-tertiary-light)",
        color: "#fff",
        border: "1px solid #fff",
        position: "relative",
        zIndex: 1,
        cursor: "pointer",
        flexShrink: 0,

        "@media only screen and (max-width: 400px)": {
          width: "auto",
          maxWidth: 280,
        },

        "&:before": {
          content: "''",
          width: "100%",
          height: "70%",
          background: "var(--color-tertiary-dark)",
          position: "absolute",
          borderRadius: "0 0 50% 50%",
          top: 0,
          left: 0,
          zIndex: -1,
        },
      }}
    >
      <CardContent>
        <Box
          sx={{ display: "flex", justifyContent: "space-between", zIndex: 5 }}
        >
          <Typography
            id="title"
            variant="h5"
            sx={{ fontSize: 18, marginBottom: 3 }}
            gutterBottom
          >
            {title}
          </Typography>
          <Typography
            id="date"
            variant="h6"
            sx={{
              fontSize: 16,
              color: "var(--color-base)",
            }}
          >
            {date}
          </Typography>
        </Box>
        <Typography id="content" sx={{ fontSize: 14, width: "75%" }}>
          {content}
        </Typography>
      </CardContent>
    </Card>
  );
};

export default NoteElement;
