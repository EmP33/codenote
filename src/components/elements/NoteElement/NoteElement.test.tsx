import { render, screen } from "@testing-library/react";
import NoteElement from "./NoteElement";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import "@testing-library/jest-dom";

describe("NoteElement Component", () => {
  const history = createMemoryHistory();
  render(
    <Router location={history.location} navigator={history}>
      <NoteElement
        hoverOnNotes={true}
        title="Flight Details"
        date="5/3/22"
        content="Get to the airport by 7am before takeoff"
        id="example"
      />
    </Router>
  );
  it("should render title, date and content properly", () => {
    expect(screen.getByText("Flight Details")).toBeInTheDocument();
    expect(screen.getByText("5/3/22")).toBeInTheDocument();
    expect(
      screen.getByText("Get to the airport by 7am before takeoff")
    ).toBeInTheDocument();
  });
});
