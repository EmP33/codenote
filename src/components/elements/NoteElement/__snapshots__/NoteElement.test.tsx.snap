// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteElement Component should render NoteElement properly 1`] = `
<ForwardRef(Card)
  sx={
    Object {
      "&:before": Object {
        "background": "var(--color-tertiary-dark)",
        "borderRadius": "0 0 50% 50%",
        "content": "''",
        "height": "70%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": -1,
      },
      "@media only screen and (max-width: 400px)": Object {
        "maxWidth": 280,
        "width": "auto",
      },
      "background": "var(--color-tertiary-light)",
      "border": "1px solid #fff",
      "color": "#fff",
      "cursor": "pointer",
      "flexShrink": 0,
      "height": 250,
      "margin": "2rem auto",
      "position": "relative",
      "width": 350,
      "zIndex": 1,
    }
  }
>
  <ForwardRef(CardContent)>
    <ForwardRef(Box)
      sx={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
          "zIndex": 5,
        }
      }
    >
      <ForwardRef(Typography)
        gutterBottom={true}
        id="title"
        sx={
          Object {
            "fontSize": 18,
            "marginBottom": 3,
          }
        }
        variant="h5"
      >
        Flight Details
      </ForwardRef(Typography)>
      <ForwardRef(Typography)
        id="date"
        sx={
          Object {
            "color": "var(--color-base)",
            "fontSize": 16,
          }
        }
        variant="h6"
      >
        5/3/22
      </ForwardRef(Typography)>
    </ForwardRef(Box)>
    <ForwardRef(Typography)
      id="content"
      sx={
        Object {
          "fontSize": 14,
          "width": "75%",
        }
      }
    >
      Get to the airport by 7am before takeoff
    </ForwardRef(Typography)>
  </ForwardRef(CardContent)>
</ForwardRef(Card)>
`;
